analyses/upstream-analysis/04_run_filter_object.Rmd

```diff
@@ -0,0 +1,404 @@
+scDblFinder_file <- file.path(scDblFinder_results_dir, "merged_seurat_obj_with_doublets.rds")
+metadata_file <- file.path(metadata_dir, "project_metadata.tsv") # metadata input file
+gradient_palette_file <- file.path(root_dir, "figures", "palettes", "gradient_color_palette.tsv")
+
+# Create results_dir
+results_dir <- file.path(module_results_dir, "04_Filter_object")
+if (!dir.exists(results_dir)) {
+  dir.create(results_dir)}
+
+# Create plots directory
+plots_dir <- file.path(module_plots_dir, "04_Filter_object") 
+if (!dir.exists(plots_dir)) {
+  dir.create(plots_dir)}
+
+source(paste0(root_dir, "/figures/scripts/theme_plot.R"))
+source(paste0(analysis_dir, "/util/function-process-Seurat.R"))
+source(paste0(analysis_dir, "/util/function-create-UMAP.R"))
+```
+
+# Read metadata file and define `sample_name`
+
+```{r read-metadata-define-sample-name, echo=TRUE}
+# Read metadata
+project.metadata <- read.csv(metadata_file, sep = '\t', header = TRUE)
+sample_name <- unique(as.character(project.metadata$ID))
+sample_name <- sort(sample_name, decreasing = FALSE)
+print(sample_name)
+```
+
+```{r define-parameters-for-plots, echo=TRUE}
+# Read color palette
+gradient_palette_df <- readr::read_tsv(gradient_palette_file, guess_max = 100000, show_col_types = FALSE) 
+```
+
+# Read and merge `seurat_qc` object
+First, we will merge the `seurat_obj.rds` objects from all samples in the project as generated by the `seurat_qc` pipeline.
+
+```{r read-merge-objects, echo=TRUE}
+# Create list 
+seurat_obj_list <- list()
+
+for (i in seq_along(sample_name)) {
+  cat("Reading in data", sample_name[i], "\n")
+  seurat_obj_list[i] <- readRDS(seurat_results_file[i])}
+print(seurat_obj_list)
+
+# Merge all seurat objects in the list into one 
+# The first parameter of merge should be a Seurat object, the second (y) can be one Seurat object or a list of several.
+seurat_obj <- merge(x = seurat_obj_list[[1]], y = seurat_obj_list[-1],
+                    project = PROJECT_NAME)
+print(DefaultAssay(seurat_obj) <- "RNA") # Switch the default to RNA
+saveRDS(seurat_obj, file = paste0(results_dir, "/", "seurat_obj_merged.rds")) # Save merged object
+#############################################################################
+# Estimate `cells_number_seurat_qc` per sample_name
+cells_number_seurat_qc <- table(seurat_obj@meta.data$orig.ident) %>%
+    as.data.frame() %>% 
+    mutate(sample_name = Var1,
+           cells_number_seurat_qc = Freq) %>% 
+    dplyr::select(-Var1, -Freq)
+```
+
+# Filter ambient RNA
+Next, we will use the corrected matrices from all samples as generated by [SoupX](https://cran.r-project.org/web/packages/SoupX/vignettes/pbmcTutorial.html). This is an optional step, so it will be skipped or not as defined in `params`. 
+
+```{r filter-ambient-RNA, echo=TRUE}
+# Indicates whether or not to use the SoupX corrected matrix for filtering or to skip it
+if (use_SoupX_filtering == "YES"){
```

Comment:
Similar to with doublets, I noticed that the parameter for SoupX filtering in the yaml file is named as `use_SoupX_filtering_filter_object` while `use_SoupX_filtering` is used in this script. Again, just wanted to make sure that is correct and won't affect how the script runs.